homeassistant/components/climate/__init__.py

```diff
@@ -42,6 +42,13 @@
 STATE_COOL = 'cool'
 STATE_IDLE = 'idle'
 STATE_AUTO = 'auto'
+STATE_BOOST = 'boost'
+STATE_AWAY = 'away'
+STATE_MANUAL = 'manual'
+STATE_COMFORT = 'comfort'
+STATE_LOWERING = 'lowering'
+STATE_VACATION = 'vacation'
+STATE_HEAT_COOL = 'heat-cool'
```

Comment:
That is to platform specific. We also use every time `_`. Pump that back to platform.

Reply:
Just the "heat-cool" or all four of them?

Reply:
No, we let it stay on the platform. If we have a other platform they will use the same value, we can move it up

Reply:
So it is important that there are no platform specific states. Otherwise we will be unable to make a frontend. Think of it as an enum in Java.

I think the problem that we are seeing here is that we are conflating concerns. We are trying to store the following things all as current operation:

 - operating state (heating, cooling, idle)
 - mode (auto, manual, preset)
 - source (electric, gas, fan_only)
 - preset (vacation, comfort, etc). Does not have to be standardized in climate component

I think that the latter 3 should be extracted as their own attributes. We can then force a limited amount of values for operation, mode and source. Preset can just be whatever the platform is spitting out.

We should also standardize:
 - `heat-cool` should become `auto`

Then there are some weird things like `dry` or `lowering`, for which I don't know what it means

Reply:
And then every platform would need to map the predefined states / modes / … to the platform specific value (if available)?